crates/bitwarden-vault/src/tree.rs

```diff
@@ -0,0 +1,275 @@
+        let item = self.items.iter().find(|i| i.data.id() == Some(tree_item_id));
+
+        if let Some(item) = item {
+            let node = self.nodes.get(item.id)?;
+
+            // Get the parent if it exists
+            let parent = node.parent_idx
+                .and_then(|pid| self.nodes.get(pid));
+
+            // Get all children nodes
+            let children: Vec<&TreeNode> = node.children_idx.iter()
+                .filter_map(|&child_id| self.nodes.get(child_id))
+                .collect();
+
+            // Get corresponding items
+            let parent_item = parent.and_then(|p| self.items.get(p.id));
+            let children_items: Vec<&TreeIndex<T>> = children.iter()
+                .filter_map(|child| self.items.get(child.id))
+                .collect();
+
+            return Some(NodeItem {
+                item: item.data.clone(),
+                parent: parent_item.map(|p| p.data.clone()),
+                children: children_items.iter().map(|i| i.data.clone()).collect()
+            })
+        }
+
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use uuid::Uuid;
+    use crate::tree::TreeItem;
+    
+    #[derive(Clone, Debug)]
+    pub struct TestItem {
+        pub id: Uuid,
+        pub name: String
+    }
+    
+    impl TreeItem for TestItem {
+        fn id(&self) -> Option<Uuid> {
+            Some(self.id)
+        }
+
+        fn short_name(&self) -> &str { self.path().last().unwrap() }
+
+        fn path(&self) -> Vec<&str> {
+            self.name
+                .split(Self::DELIMITER)
+                .filter(|s| !s.is_empty())
+                .collect::<Vec<&str>>()
+        }
+
+        const DELIMITER: char = '/';
+    }
+    
+    #[test]
+    fn given_collection_with_one_parent_and_two_children_when_getting_parent_then_parent_is_returned_with_children_and_no_parent() {
+        let parent_id = Uuid::new_v4();
+        let mut items = vec![
+            TestItem {
+                id: Uuid::new_v4(),
+                name: "parent/child1".to_string()
+            },
+            TestItem {
+                id: parent_id,
+                name: "parent".to_string()
+            },            
+            TestItem {
+                id: Uuid::new_v4(),
+                name: "parent/child2".to_string()
+            },
+        ];
+        
+        let node_option = Tree::from_items(&mut items)
+            .get_item_by_id(parent_id);
+        
+        if let Some(node) = node_option {
+            let item = node.item;
+            let parent = node.parent;
+            let children = node.children;
+            
+            assert_eq!(children.len(), 2);
+            assert_eq!(item.id(), Some(parent_id));
+            assert_eq!(item.short_name(), "parent");
+            assert_eq!(item.path(), ["parent"]);
+            assert!(parent.is_none());
+        } else {
+            panic!("Node not found");
+        }
```

Comment:
suggestion: simplify by expecting a value

```suggestion
        let node = Tree::from_items(&mut items)
            .get_item_by_id(parent_id)
            .expect("Node not found");
        
        let item = node.item;
        let parent = node.parent;
        let children = node.children;
            
        assert_eq!(children.len(), 2);
        assert_eq!(item.id(), Some(parent_id));
        assert_eq!(item.short_name(), "parent");
        assert_eq!(item.path(), ["parent"]);
        assert!(parent.is_none());
```